AISKU/Tests/TestFramework/TestClass.ts

```diff
@@ -1,23 +1,74 @@
-/// <reference path="../../../common/Tests/External/sinon.d.ts" />
 /// <reference path="../../../common/Tests/External/qunit.d.ts" />
-/// <reference path="Assert.ts" />
-/// <reference path="./TestCase.ts"/>
+import { SinonSandbox, SinonSpy, SinonStub, SinonMock, SinonFakeXMLHttpRequest } from 'sinon';
+import * as sinon from 'sinon';
+import dynamicProto from '@microsoft/dynamicproto-js';
+import { ITestContext, StepResult, TestCase, TestCaseAsync } from './TestCase';
+import { Assert } from './Assert';
+import { CoreUtils, getGlobal } from '@microsoft/applicationinsights-core-js';
+
+const stepRetryCnt = 'retryCnt';
+
+export interface FakeXMLHttpRequest extends XMLHttpRequest {
```

Comment:
Updated the TestClass with extra mocking ability from other projects.

Also added the ability to now emulate an ES3 environment using a simple boolean during class initialization